packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Joe Adams <adams10301@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: pulp_repo
+author: "Joe Adams <adams10301@gmail.com> github.com/sysadmind"
+short_description: Add or remove Pulp repos from a remote host.
+description:
+   - Add or remove Pulp repos from a remote host.
+version_added: "2.1"
+requirements: []
+options:
+  name:
+    description:
+      - name of the repo to add or remove (correlates to pulp's repo-id)
+    required: true
+    default: null
+  state:
+    description:
+      - The repo state
+    required: false
+    default: present
+    choices: [ "present", "absent" ]
+  repo_type:
+    description:
+      - Repo plugin type to use (i.e. rpm, docker)
```

Comment:
Not fully aware of pulp, is it possible to give the user a choice list for all possible repo types?